### PR TITLE
Alignfix

### PR DESCRIPTION
**Pull request by DrLynch**
Fixes the doc alignment problem.  Builds on the underscore fix.

### CODE REVIEW

File: learning_observer/learning_observer/dashboard.py [hunk 1]
```diff
@@ -181,6 +181,7 @@ async def student_state_fetcher():
         '''
         students = []
         for student in roster:
+
```
Comment (pmitros) [inline learning_observer/learning_observer/dashboard.py:184]: No newline.

File: learning_observer/learning_observer/dashboard.py [hunk 2]
```diff
@@ -297,8 +308,8 @@ async def websocket_dashboard_view(request):
     # the same data format for 1 student as for a classroom of
     # students.
     if student_id is not None:
-        roster = [r for r in roster if r['userId'] == student_id]
-
+        roster = [r for r in roster if r['user_id'] == student_id]
+
```
Comment (pmitros) [inline learning_observer/learning_observer/dashboard.py:312]: No whitespace.

File: learning_observer/learning_observer/dashboard.py [hunk 3]
```diff
@@ -312,6 +323,7 @@ async def websocket_dashboard_view(request):
 
     while True:
         sd = await student_state_fetcher()
+
```
Comment (pmitros) [inline learning_observer/learning_observer/dashboard.py:326]: No newline.

File: learning_observer/learning_observer/rosters.py [hunk 4]
```diff
@@ -50,6 +74,8 @@
 import learning_observer.kvs
 import learning_observer.log_event as log_event
 import learning_observer.paths as paths
+import learning_observer.auth as auth
+
```
Comment (pmitros) [inline learning_observer/learning_observer/rosters.py:78]: No extra newline.

File: learning_observer/learning_observer/rosters.py [hunk 5]
```diff
@@ -59,7 +85,71 @@
...
+}
+    
+
+
+# d = {
+#    "descriptionHeading": "foo",
+#    "subdict": {
+#       "calendarId": "bar"
+#    }
+# }
```
Comment (pmitros) [inline learning_observer/learning_observer/rosters.py:128]: Please remove the comment (it's okay to have it as data in a test case (e.g. docstring or `if __name__ == '__main__':`

File: learning_observer/learning_observer/rosters.py [hunk 6]
```diff
@@ -59,7 +85,71 @@
...
+        
+    for k, v in list(d.items()):
+
+        if k in translations:
+            d[translations[k]] = d.pop(k)
+        else: print("UNTRANSLATED KEY: ", k)
+
+        if isinstance(v, dict):
+            camel_to_underscore_rename_dict(v, translations)
+
```
Comment (pmitros) [inline learning_observer/learning_observer/rosters.py:148]: Way too much whitespace.

File: learning_observer/learning_observer/rosters.py [hunk 7]
```diff
@@ -85,13 +176,65 @@ def clean_google_ajax_data(resp_json, key, sort_key, default=None):
...
+    # have a list of student dicts in resp_json.
+    if (key == 'students'): 
+        adjust_external_gc_ids(resp_json)
+         
     # Sort the list
     if sort_key is not None:
         resp_json.sort(key=sort_key)
-
+  
+
```
Comment (pmitros) [inline learning_observer/learning_observer/rosters.py:192]: Too much whitespace.

File: learning_observer/learning_observer/rosters.py [hunk 8]
```diff
@@ -196,7 +340,7 @@ async def synthetic_ajax(
     else:
         debug_log("Roster data source is not recognized:", settings.settings['roster_data']['source'])
         raise ValueError("Roster data source is not recognized: {}".format(settings.settings['roster_data']['source'])
-                         + " (should be 'test' or 'filesystem')")
+                            + " (should be 'test' or 'filesystem')")
```
Comment (pmitros) [inline learning_observer/learning_observer/rosters.py:343]: No random formatting changes, please.

File: learning_observer/learning_observer/rosters.py [hunk 9]
```diff
@@ -228,6 +372,7 @@ async def google_ajax(
     this error back to the JavaScript client, which can then handle
     loading the auth page.
     '''
+
```
Comment (pmitros) [inline learning_observer/learning_observer/rosters.py:375]: No whitespace.

File: learning_observer/prototypes/dash_wo_teacher_dashboard/wo_dash/module.py [hunk 10]
```diff
@@ -49,3 +49,13 @@
         "666e8a2e34aae8e79904f8e9b4c0ccae40249897cce5f5ae58d12cc1b3985e588d6"
     }
 }
+
```
Comment (pmitros) [inline learning_observer/prototypes/dash_wo_teacher_dashboard/wo_dash/module.py:52]: One too many blank lines.

File: modules/writing_observer/writing_observer/aggregator.py [hunk 11]
```diff
@@ -134,21 +144,44 @@ async def get_latest_student_documents(student_data):
 
     kvs = learning_observer.kvs.KVS()
 
+    
+    # Compile a list of the active students defined as those
+    # for which we have some "last_document".  
+    active_students = [s for s in student_data if 'writing_observer.writing_analysis.last_document' in s]
+
```
Comment (pmitros) [inline modules/writing_observer/writing_observer/aggregator.py:151]: One line.

File: modules/writing_observer/writing_observer/aggregator.py [hunk 12]
```diff
@@ -134,21 +144,44 @@ async def get_latest_student_documents(student_data):
...
-                KeyField.STUDENT: s['userId'],
+                KeyField.STUDENT: s['user_id'],
                 EventField('doc_id'): s['writing_observer.writing_analysis.last_document']['document_id'] 
             },
             KeyStateType.INTERNAL
-        ) for s in student_data])
-
-    writing_data = await kvs.multiget(keys=document_keys)
+        ) for s in active_students]) # in student_data if 'writing_observer.writing_analysis.last_document' in s])
+
```
Comment (pmitros) [inline modules/writing_observer/writing_observer/aggregator.py:163]: No spaces on empty line.

File: modules/writing_observer/writing_observer/aggregator.py [hunk 13]
```diff
@@ -134,21 +144,44 @@ async def get_latest_student_documents(student_data):
...
+        student = active_students[idx]
+        doc     = kvs_data[idx]
+        
+        # If we have an empty item we simply return an empty dict with the
+        # student but an empty doc value.
+        if (doc is None): doc = {}
+
+        # Now insert the student data and pass it along.
+        doc['student'] = student
+        writing_data.append(doc)
```
Comment (pmitros) [inline modules/writing_observer/writing_observer/aggregator.py:183]: This should be done as a list comprehension. E.g.:
`[make_item(student, doc) for (student, doc) in zip(active_students, kvs_data]`

This is more readable, and avoids `append`ing to a linked list.

File: modules/writing_observer/writing_observer/aggregator.py [hunk 14]
```diff
@@ -167,6 +200,7 @@ async def merge_with_student_data(writing_data, student_data):
     '''
     Add the student metadata to each text
     '''
+
```
Comment (pmitros) [inline modules/writing_observer/writing_observer/aggregator.py:203]: No newlines

File: modules/writing_observer/writing_observer/aggregator.py [hunk 15]
```diff
@@ -179,16 +213,35 @@ async def merge_with_student_data(writing_data, student_data):
...
+    
+    This code needs to take the student data as a dict and then 
+    collect the latest writing data for each student (assuming 
+    they have it). The code then passes that writing data on 
+    to Paul's code for processing.  For the time being this 
+    works by essentially building up some large dicts that 
+    contain the text and student data together.  
+
+    In the long run this should *all* be replaced by a cleaner
+    object interface that hides some of this from the user
```
Comment (pmitros) [inline modules/writing_observer/writing_observer/aggregator.py:225]: Object interface? I'd kill the word 'object'.

File: modules/writing_observer/writing_observer/aggregator.py [hunk 16]
```diff
@@ -179,16 +213,35 @@ async def merge_with_student_data(writing_data, student_data):
 async def latest_data(student_data):
     '''
     HACK HACK HACK
-
-    I just hardcoded this, breaking abstractions, repeating code, etc.
+    
+    This code needs to take the student data as a dict and then 
+    collect the latest writing data for each student (assuming 
+    they have it). The code then passes that writing data on 
+    to Paul's code for processing.  For the time being this
```
Comment (pmitros) [inline modules/writing_observer/writing_observer/aggregator.py:220]: I'd consider removing names (Paul's code, CL, etc.). Readers might not know what this is referring to.

**Comment by bradley-erickson** [comment]
This pull request is now stale. Multiple pull requests with the same code were opened up. This results in future pull requests showing those items as changes since they are made with a different version of Master. This is messy to deal with. The avoid this in the future **specify different features to different pull requests**.

**Pull request closed without merging**